### ISSUE TITLE
poetry_core needs poetry to run tests

### ISSUE THREAD

**Issue by mtelka** [python-poetry/poetry#7826]
The `tox.ini` file for poetry_core says that the command to run tests is `poetry`, but such command is not provided by any listed dependencies in neither `tox.ini` file (actually, there is no `deps` for `[testenv]`), nor `pyproject.toml` file.  I assume the `poetry` command is provided by the `poetry` project, so `poetry` should be listed in `tox.ini` as `deps` in `[testenv]`.

**Comment by dimbleby** [comment]
it doesn't.  The fragment of tox configuration that you have found seems to be dead code.

I expect an MR cleaning it up would be welcome, though per #7824 I'd likely encourage an MR that went even further and removed tox completely from poetry-core

**Comment by mtelka** [comment]
Then... how should a downstream distribution run tests for poetry_core?

**Comment by dimbleby** [comment]
Do you believe that `tox.ini` is mandatory for a project to be included in a downstream distribution?  Plainly this is not so.  

Create a virtual environment, install the product, run the tests.  Just like what poetry's own pipelines do, in fact.

**Comment by mtelka** [comment]
> Do you believe that `tox.ini` is mandatory for a project to be included in a downstream distribution? Plainly this is not so.

Of course, it is not mandatory, but if it is included then it is expected `tox` is used to run tests, so `tox.ini` should work. :-)

> Create a virtual environment, install the project, run the tests. Just like what poetry's own pipelines do, in fact.

Sorry, I cannot create virtual environment because I package poetry_core for an operating system where the resulting package will run in real system (current environment), not in a virtual environment.

Anyway, you said: "run the tests".  That's exactly my question: how to run test?  How is poetry's own pipelines running tests?  Where can I find more info about how to run poetry tests?  Thank you.

**Comment by dimbleby** [comment]
> if it is included... `tox.init` should work

sure, we just have different answers to how to resolve that - tox is in fact not used, so remove it!

>  How is poetry's own pipelines running tests?

https://github.com/python-poetry/poetry/tree/main/.github/workflows, and similar for poetry-core

**Comment by mtelka** [comment]
> > How is poetry's own pipelines running tests?
> 
> https://github.com/python-poetry/poetry/tree/master/.github/workflows, and similar for poetry-core

That's... just bunch of yaml/yml files. :-(

Could you please tell me what command(s) should I run to test poetry_core?  Without github, on a standalone operating system.

Thank you.

**Comment by dimbleby** [comment]
if you read the yaml files you will see that the steps are basically just executing commands.  Run those commands.  

As far as I know that's as good as the documentation gets.

**Comment by mtelka** [comment]
Unfortunately, this is not very reliable.  Always when you change any yaml file I'd need to analyze the change and adapt our test recipe.  That's very inconvenient.  It is sad to see that the poetry/poetry_core project creates so high barrier for downstream packagers to get your project properly tested :-(.

**Comment by kapilt** [comment]
reading through the yaml files its mostly just running pytest, its a little bit unclear what your trying to do.

**Comment by finswimmer** [comment]
> Could you please tell me what command(s) should I run to test poetry_core

Essentially you have to:

* install `poetry-core` with its dependencies
* install the dependency listed in the `test` group
* run `pytest` on the `tests` folder

How you do this, depends really on your usecase. There is no "one-thing-that-always-work" that Poetry can provide here. In case you want to test during development it is a simple `poetry install` followed by a `poetry run pytest tests`.
